### PR TITLE
[[ Bug 19668 ]] Fixed crash when accessing ImageData

### PR DESCRIPTION
**Pull request by abrisan**

### CODE REVIEW

**Comment by runrevmark** [comment]
@livecodealex : Could you explain why this fixes the crash? Or, more specifically, what was causing it when the image was locked before?

File: engine/src/exec-interface-image.cpp [hunk 1]
```diff
@@ -431,45 +431,68 @@ void MCImage::SetText(MCExecContext& ctxt, MCDataRef p_text)
...
+        else
+        {
             // SN-2014-01-31: [[ Bug 11462 ]] Opening an image to get its data should not
             // reset its size: F_LOCK_LOCATION ensures the size - and the location, which
             // doesn't matter here - are read as they are stored.
             bool t_tmp_locked;
             t_tmp_locked = false;
             
-            if (!getflag(F_LOCK_LOCATION))
+            if (!t_locked)
```
Comment (runrevmark) [inline engine/src/exec-interface-image.cpp]: I think we should be able to make this:
```
MCImageBitmap *t_bitmap = nullptr;
bool t_tmp_locked_location = false;
if (m_rep->GetType() == kMCImageRepMutable)
{
  t_success = static_cast<MCMutableImageRep *>(m_rep)->LockBitmap(0, 0, t_bitmap);
}
else
{
  -- make sure locklocation is set
  -- make sure image is open
  t_success = lockbitmap(t_bitmap, false);
}

... make image data ...

if (m_rep->GetType() == kMCImageRepMutable)
{
  UnlockBitmap
}
else
{
  unlockbitmap
  closeimage
  reset locklocation
}
```
The rationale here is that if the image has a mutableimagerep, then we don't need to open or do anything else, as we are using the bitmap direct from the mutableimagerep (which will always have one).

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 411709a

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 411709ab897b72ee6ca684b4531d4d245a396ebf

**Comment by livecode-vulcan** [comment]
😞 test failure 411709ab897b72ee6ca684b4531d4d245a396ebf

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/389)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/397)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/376)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/319)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/319)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/384)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/389)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/387)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/394)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/387)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/394)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/413)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/383)

**Comment by runrevmark** [comment]
@livecode-vulcan review ok a34f3c3

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok a34f3c36e3608c9953a43eb773e8bad9275387a2

**Comment by livecode-vulcan** [comment]
😎 test success a34f3c36e3608c9953a43eb773e8bad9275387a2

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/391)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/399)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/378)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/321)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/321)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/386)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/391)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/389)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/396)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/389)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/396)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/415)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/385)

**Pull request merged**